go-controller/pkg/factory/handler.go

```diff
@@ -48,6 +48,19 @@ type Handler struct {
 	// example: a handler with priority 0 will process the received event first
 	// before a handler with priority 1.
 	priority int
+
+	// processedInitialAdd used to determine if this handler has ever processed this object
+	// before attempting to handle any deletes/updates
+	processedInitialAdd *sync.Map
```

Comment:
Let's see what is the price to pay in memory consumption here. Why a pointer though?